### DISCUSSION TITLE
brew installation internal github repo returns 404

### DISCUSSION THREAD

**Discussion by trent-abc** [community/community#54280] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

So this was working about a week ago, and now it isn't (404). No changes on my side.

I have an internal repo, homebrew/tap and it points to two other repos tar.gz for installing brew packages.  Manually putting these links in my browser return a 200 with the downloaded file.  I have tried recreating a PAT and using the new HOMEBREW_GITHUB_API_TOKEN but to no benefit.  Assuming something broke on the github side or while upgrading my mac but unsure where the issue lies.

Context:

~/.netrc
```
machine github.com
  login my-github-username
  password your_token_here
```

~/.zshrc
```
export HOMEBREW_GITHUB_API_TOKEN=your_token_here
```

See what's available on tap - Currently working
```
brew tap MyCompany/tap
brew tap-info MyCompany/tap --json | jq -r '.[].formula_names | .[]'
MyCompany/tap/aws-sso-login
```

Install the tools - Currently failing
```
brew install aws-sso-login

...

Error: Failed to download resource "aws-sso-login"
Failure while executing; `/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.16-6-ge62a839\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 13.3.1\)\ curl/7.87.0 --header Accept-Language:\ en --retry 3 --fail --location --silent --head --request GET https://github.com/MyCompany/ccc/releases/download/0.2.0/aws-sso-login.0.2.0.linux-amd64.tar.gz` exited with 22. Here's the output:
curl: (22) The requested URL returned error: 404
HTTP/2 404
server: GitHub.com
date: Tue, 02 May 2023 18:25:31 GMT
content-type: text/plain; charset=utf-8
vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With
cache-control: no-cache
strict-transport-security: max-age=31536000; includeSubdomains; preload
x-frame-options: deny
x-content-type-options: nosniff
x-xss-protection: 0
referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
content-security-policy: default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'
content-length: 9
x-github-request-id: C60F:1AC5:5A2453:883FF8:6451559B
```

**Comment by mrnvzch** [comment]
@trent-abc did you figure out a workaround? I find it hard to believe that this bug has existed since May and so few people have complained.

**Comment by trent-abc** [comment]
Thanks for checking in. With 8 upvotes it seems like other people also have the same issue. definitely something to be investigated, have you tried to reproduce with an internal repo?

```
curl: (22) The requested URL returned error: 404
HTTP/2 404
server: GitHub.com
date: Tue, 15 Aug 2023 17:47:13 GMT
```

**Comment by mrnvzch** [comment]
Yeah I am getting the same problem. My point is that I'd expect more people to be hosting private formulas than just 8 people so I'm slightly taken aback by the lack of activity on this, or mention of this issue anywhere else. I still don't know if this is a regression on brew's part or a change in the github api, nor do I have time to figure it out. For the time being I'm just hosting tooling in CodeArtifact instead while I await a fix or a workaround.

**Comment by mrnvzch** [comment]
@trent-abc fyi - I did some more searching - [here](https://github.com/Homebrew/brew/issues/15169#issuecomment-1504229816) is a workaround.

**Comment by trent-abc** [comment]
@mrnvzch Life saver, thanks a lot man this thread was golden. In other words, they know it's broken and can't fix it.  This comment saved my life:

https://github.com/Homebrew/brew/issues/15169#issuecomment-1500653530

In order to avoid link rot I'll just paste it here for those curious

Modify `Formula/lib/private.rb`, find class `GitHubPrivateRepositoryReleaseDownloadStrategy` and add the following section:

```
  def resolve_url_basename_time_file_size(url, timeout: nil)
    [download_url, "", Time.now, 0, false]
  end
```
